File: AD-HOC/1107/main.py
def num_lasers(a, c, x):
    lasers = a - x[0]
    for i in range(1, c):
        if x[i-1] > x[i]:
            lasers += x[i-1] - x[i]
    return lasers


a, c = map(int, input().split())
while a or c:
    x = list(map(int, input().split()))
    print(num_lasers(a, c, x))
    a, c = map(int, input().split())

File: AD-HOC/1107/main.dart
import 'dart:io';

int numLasers(int a, int c, List<int> x) {
  int lasers = a - x.first;
  for (int i = 1; i < c; i++) {
    if (x[i - 1] > x[i]) lasers += x[i - 1] - x[i];
  }
  return lasers;
}

void main() {
  var [a, c] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  while (a > 0 || c > 0) {
    List<int> x = stdin.readLineSync()!.split(' ').map(int.parse).toList();
    print(numLasers(a, c, x));
    [a, c] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  }
}
